### ISSUE TITLE
[ModelNet40]Per-class IoU/Accuracy all zero for many classes

### ISSUE THREAD

**Issue by jinnuqy3** [Pointcept/Pointcept#462]
I’m training the cls-spunet-v1m1-0-base.py model on ModelNet40 , and using the version 1.6.0, but several object categories end up with IoU=0 and Acc=0, pulling down my overall accuracy. And the cls-ptv3-v1m1-0-base.py also has same issue.

What common pitfalls can cause entire categories to receive zero predictions?

[2025-05-15 14:03:47,428 INFO evaluator.py line 76 8861] Val result: mIoU/mAcc/allAcc 0.4709/0.5991/0.7561.
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_0-airplane Result: iou/accuracy 0.8929/1.0000
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_1-bathtub Result: iou/accuracy 0.6545/0.7200
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_2-bed Result: iou/accuracy 0.8774/0.9300
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_3-bench Result: iou/accuracy 0.3077/0.4000
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_4-bookshelf Result: iou/accuracy 0.5500/0.9900
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_5-bottle Result: iou/accuracy 0.8496/0.9600
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_6-bowl Result: iou/accuracy 0.0000/0.0000
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_7-car Result: iou/accuracy 0.8868/0.9400
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_8-chair Result: iou/accuracy 0.8636/0.9500
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_9-cone Result: iou/accuracy 0.6333/0.9500
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_10-cup Result: iou/accuracy 0.0000/0.0000
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_11-curtain Result: iou/accuracy 0.3393/0.9500
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_12-desk Result: iou/accuracy 0.3261/0.3488
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_13-door Result: iou/accuracy 0.0000/0.0000
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_14-dresser Result: iou/accuracy 0.3763/0.8140
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_15-flower_pot Result: iou/accuracy 0.0000/0.0000
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_16-glass_box Result: iou/accuracy 0.5904/0.9800
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_17-guitar Result: iou/accuracy 0.8300/0.8300
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_18-keyboard Result: iou/accuracy 0.6923/0.9000
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_19-lamp Result: iou/accuracy 0.3077/0.6000
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_20-laptop Result: iou/accuracy 0.5714/1.0000
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_21-mantel Result: iou/accuracy 0.8333/0.8500
[2025-05-15 14:03:47,429 INFO evaluator.py line 82 8861] Class_22-monitor Result: iou/accuracy 0.9700/0.9700
[2025-05-15 14:03:47,430 INFO evaluator.py line 82 8861] Class_23-night_stand Result: iou/accuracy 0.1262/0.1512
[2025-05-15 14:03:47,430 INFO evaluator.py line 82 8861] Class_24-person Result: iou/accuracy 0.0000/0.0000
[2025-05-15 14:03:47,430 INFO evaluator.py line 82 8861] Class_25-piano Result: iou/accuracy 0.7048/0.7400
[2025-05-15 14:03:47,430 INFO evaluator.py line 82 8861] Class_26-plant Result: iou/accuracy 0.6838/0.9300
[2025-05-15 14:03:47,430 INFO evaluator.py line 82 8861] Class_27-radio Result: iou/accuracy 0.0000/0.0000
[2025-05-15 14:03:47,430 INFO evaluator.py line 82 8861] Class_28-range_hood Result: iou/accuracy 0.8762/0.9200
[2025-05-15 14:03:47,430 INFO evaluator.py line 82 8861] Class_29-sink Result: iou/accuracy 0.2727/0.6000
[2025-05-15 14:03:47,430 INFO evaluator.py line 82 8861] Class_30-sofa Result: iou/accuracy 0.9245/0.9800
[2025-05-15 14:03:47,430 INFO evaluator.py line 82 8861] Class_31-stairs Result: iou/accuracy 0.0000/0.0000
[2025-05-15 14:03:47,430 INFO evaluator.py line 82 8861] Class_32-stool Result: iou/accuracy 0.0000/0.0000
[2025-05-15 14:03:47,430 INFO evaluator.py line 82 8861] Class_33-table Result: iou/accuracy 0.7015/0.9400
[2025-05-15 14:03:47,430 INFO evaluator.py line 82 8861] Class_34-tent Result: iou/accuracy 0.3667/0.5500
[2025-05-15 14:03:47,430 INFO evaluator.py line 82 8861] Class_35-toilet Result: iou/accuracy 0.9900/0.9900
[2025-05-15 14:03:47,430 INFO evaluator.py line 82 8861] Class_36-tv_stand Result: iou/accuracy 0.2696/0.3100
[2025-05-15 14:03:47,430 INFO evaluator.py line 82 8861] Class_37-vase Result: iou/accuracy 0.5662/0.7700
[2025-05-15 14:03:47,430 INFO evaluator.py line 82 8861] Class_38-wardrobe Result: iou/accuracy 0.0000/0.0000
[2025-05-15 14:03:47,430 INFO evaluator.py line 82 8861] Class_39-xbox Result: iou/accuracy 0.0000/0.0000

**Comment by aldinorizaldy** [comment]
Hi @jinnuqy3 , did you get some explanations about your results?

I got similar results with my own dataset for object classification with more than 30 classes. In my observation, classes with extremely low samples are very likely to get zero accuracy. Quickly check on the statistic of ModelNet40, each class with zero accuracy on your results have about very low ratio of the total samples. Compared to major classes, these minor classes suffer from imbalance problem. However, I notice that not every class with low samples got zero accuracy on your results and I found this is interesting to investigate. 

I plan to include class weights in the loss function, by passing the inverse class weights in the config `criteria=[dict(type="CrossEntropyLoss", loss_weight=1.0, ignore_index=-1)],` and see if that helps.

**Comment by Gofinge** [comment]
Hey guys. I think the balance loss weight for CrossEntropyLoss is a short-term solution. For the long term, we need a pre-trained backbone learned with object datasets. We are working on it now (trained one model with "all" pattern of point cloud), and will probably have some outcome by the end of the year.
